usaspending_api/reporting/v2/views/agencies/publish_dates.py

```diff
@@ -131,9 +131,14 @@ def format_results(self, result_list):
     def get(self, request):
         if "publication_date" in self.pagination.sort_key:
             self.validate_publication_sort(self.pagination.sort_key)
+            displayed_periods = list(
+                filter(lambda period: is_valid_monthly_period(self.fiscal_year, period), list(range(2, 13)))
+            )
             sort_key = deepcopy(self.pagination.sort_key)
-            # we get the index of the periods by subtracting 2, since we index from 0 and have no period 1
-            pub_sort = int(sort_key.split(",")[1]) - 2
+            try:
+                pub_sort = displayed_periods.index(int(sort_key.split(",")[1]))
```

Comment:
I see what you're doing here, but it seems a little tricky. Can you add a comment clarifying that the agencies returned from `self.get_agency_data()` have the same filtered list of periods?

Reply:
Pushed changes to make this more clear in code